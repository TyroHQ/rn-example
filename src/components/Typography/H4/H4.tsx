import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";

const H4Container = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class H4 extends React.Component<{}> {
  static defaultProps: H4["props"];
  render() {
    return <Text>H4</Text>;
  }
}
