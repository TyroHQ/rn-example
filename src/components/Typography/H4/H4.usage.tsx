import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { H4 } from "./H4";

export const basic = () => <H4 />;
