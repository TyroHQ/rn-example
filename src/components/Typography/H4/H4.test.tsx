import { shallow } from "enzyme";
import * as usage from "./H4.usage";

describe("<H4 />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
