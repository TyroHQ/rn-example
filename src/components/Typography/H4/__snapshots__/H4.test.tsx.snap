// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<H4 /> renders well in basic mode 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
>
  H4
</Text>
`;
